### PR TITLE
test(engine): add query job integration test

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #6776

### What is changed and how it works?

mimic print_status test

#### Tests 

 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Comment by lance6716** [comment]
/run-engine-integration-test

very short, ptal @amyangfei @CharlesCheung96

File: engine/test/integration_tests/dm_many_tables/run.sh [hunk 1]
```diff
@@ -19,22 +19,24 @@ function run() {
...
-		run_sql "INSERT INTO dm_many_tables.t$i VALUES (9, 90009);"
+		run_sql --quiet "INSERT INTO dm_many_tables.t$i VALUES (9, 90009);"
 	done
 
 	# create job & wait for job finished
 	job_id=$(create_job "DM" "$CUR_DIR/conf/job.yaml" "dm_many_tables")
-	exec_with_retry --count 200 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.TaskStatus.\"mysql-01\".Status.Stage == \"Finished\"'"
+	# check progress is forwarded gradually, not jump to "finished"
+	exec_with_retry --count 200 "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id/status\" | tee /dev/stderr | jq -e '.task_status.\"mysql-01\".status.status | .finishedBytes > 0 and .finishedBytes < .totalBytes'"
+	exec_with_retry "curl \"http://127.0.0.1:10245/api/v1/jobs/$job_id\" | tee /dev/stderr | jq -e '.state == \"Finished\"'"
```
Comment (amyangfei) [inline engine/test/integration_tests/dm_many_tables/run.sh:34]: L33 needs the job doesn't finish, L34 checks job is finished?

Comment (lance6716) [reply engine/test/integration_tests/dm_many_tables/run.sh:34]: yes, "check progress is forwarded gradually, not jump to finished"

**Review by amyangfei — APPROVED** [review]

File: engine/test/integration_tests/dm_many_tables/conf/job.yaml [hunk 2]
```diff
@@ -12,6 +12,7 @@ upstreams:
       password: ''
     source-id: mysql-01
     block-allow-list: balist-01
+    loader-thread: 1
```
Comment (amyangfei) [inline engine/test/integration_tests/dm_many_tables/conf/job.yaml:15]: Why adding this config item

Comment (lance6716) [reply engine/test/integration_tests/dm_many_tables/conf/job.yaml:15]: without this config lightning runs too fast so we can only see 0% directly jumped to finished 😂

**Review by CharlesCheung96 — APPROVED** [review]

**Pull request merged**